spec/json_matchers/match_response_schema_spec.rb

```diff
@@ -87,30 +87,67 @@
 
   it "does not fail when the schema matches" do
     create_schema("array_schema", {
-      "type" => "array",
-      "items" => { "type" => "string" },
+      "type": "array",
+      "items": { "type": "string" },
     })
 
     expect(response_for(["valid"])).to match_response_schema("array_schema")
   end
 
   it "supports $ref" do
-    create_schema("single", {
-      "type" => "object",
-      "required" => ["foo"],
-      "properties" => {
-        "foo" => { "type" => "string" },
+    create_schema("user", {
+      "id": "file:/#{JsonMatchers.schema_root}/user.json#",
+      "type": "object",
+      "required": ["id"],
+      "properties": {
+        "id": {
+          "type": "integer"
+        }
       }
     })
-    create_schema("collection", {
-      "type" => "array",
-      "items" => { "$ref" => "single.json" },
+    create_schema("users", {
+      "id": "file:/#{JsonMatchers.schema_root}/users.json#",
+      "type": "object",
+      "definitions": {
+        "users": {
+          "description": "A collection of users",
+          "example": [{ "id": "1" }],
+          "type": "array",
+          "items": { "$ref": "file:/#{JsonMatchers.schema_root}/user.json#" }
+        }
+      },
+      "required": ["users"],
+      "properties": { "users": { "$ref": "#/definitions/users" } }
     })
 
-    valid_response = response_for([{ "foo" => "is a string" }])
-    invalid_response = response_for([{ "foo" => 0 }])
+    valid_response = response_for({ "users": [{ "id": 1 }] })
+    invalid_response = response_for({ "users": [{ "id": "invalid" }]})
+
+    expect(valid_response).to match_response_schema("users")
+    expect(invalid_response).not_to match_response_schema("users")
+  end
+
+  it "supports the 'id' keyword" do
+    create_schema("top-level-schema", {
+      "$schema": "http://json-schema.org/draft-04/schema#",
+      "type": "object",
+      "properties": {
+        "a": { "$ref": "file:/#{JsonMatchers.schema_root}/nested.json#" }
+      }
+    })
+    create_schema("nested-schema", {
+      "$schema": "http://json-schema.org/draft-04/schema#",
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.